### PR TITLE
Fix error with variance uncertainties, fix looping bug on initial conversion

### PR DESCRIPTION
**Pull request by rosteen**
Leaving as a draft for now since I'm still trying to fix the uncertainty conversion going between wavelength and frequency space.

### CODE REVIEW

**Review by javerbukh — APPROVED** [review]
So this not only fixes the uncertainties bug that popped up, it also fixes the data flashing bug. So I think this should be fast-tracked into v1.0.2.

That said, if I run `specviz.app.get_viewer("spectrum-viewer").show_uncertainties()` when the data is first loaded, it looks normal. If I convert units, however, the shaded uncertainties area covers the entire graph. In my opinion, this is not a blocker for this PR, but I think once this is merged we should create an issue for that so that we can continue to improve the uncertainties conversion (it seems like this will be a lengthy ticket for a future sprint rather than something that can be hacked together in an afternoon, but I will defer to @rosteen on this).

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (eteq) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:152]: instead of the try/except, might it work to just directly do what's happening in the ``except`` clause?  Then in ``except`` instead you just drop the uncertainties and issue a warning/snackbar message about dropping uncertainties due to incompatible types or something

**Review by eteq — APPROVED** [review]
My inline suggestions I think might make this PR better/cleaner, but are not mandatory because I think the current solution *also* works.  So I'm approving, but @rosteen should take a quick look and change to the suggestions I made if he thinks that's a good idea.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (eteq) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:154]:
```suggestion
                    "Warning: Unrecognized uncertainty type, cannot guarantee conversion so dropping uncertainty in resulting data",
```

**Review by eteq — APPROVED** [review]
LGTM modulo a tiny wording change.

**Pull request merged**
